code/modules/mob/living/simple_animal/animals/giant_spider.dm

```diff
@@ -98,31 +99,36 @@
 	get_light_and_color(parent)
 	..()
 
-/mob/living/simple_animal/hostile/giant_spider/PunchTarget()
+/mob/living/simple_animal/hostile/giant_spider/DoPunch(var/atom/A)
 	. = ..()
-	if(isliving(.))
-		var/mob/living/L = .
-		if(L.reagents)
-			L.reagents.add_reagent(poison_type, poison_per_bite)
-			if(prob(poison_chance))
-				L << "<span class='warning'>You feel a tiny prick.</span>"
-				L.reagents.add_reagent(poison_type, poison_per_bite)
-
-/mob/living/simple_animal/hostile/giant_spider/nurse/PunchTarget()
+	if(.) // If we succeeded in hitting.
+		if(isliving(A))
+			var/mob/living/L = A
+			if(L.reagents)
+				var/target_zone = pick(BP_TORSO,BP_TORSO,BP_TORSO,BP_L_LEG,BP_R_LEG,BP_L_ARM,BP_R_ARM,BP_HEAD)
+				if(L.can_inject(src, null, target_zone))
+					L.reagents.add_reagent(poison_type, poison_per_bite)
+					if(prob(poison_chance))
+						to_chat(L, "<span class='warning'>You feel a tiny prick.</span>")
+						L.reagents.add_reagent(poison_type, poison_per_bite)
+
+/mob/living/simple_animal/hostile/giant_spider/nurse/DoPunch(var/atom/A)
 	. = ..()
-	if(ishuman(.))
-		var/mob/living/carbon/human/H = .
-		if(prob(5))
-			var/obj/item/organ/external/O = pick(H.organs)
-			if(!(O.robotic >= ORGAN_ROBOT))
-				var/eggcount
-				for(var/obj/I in O.implants)
-					if(istype(I, /obj/effect/spider/eggcluster))
-						eggcount ++
-				if(!eggcount)
-					var/eggs = new /obj/effect/spider/eggcluster/small(O, src)
-					O.implants += eggs
-					H << "<span class='warning'>The [src] injects something into your [O.name]!</span>"
+	if(.) // If we succeeded in hitting.
+		if(ishuman(A))
+			var/mob/living/carbon/human/H = A
+			if(prob(egg_inject_chance))
+				var/target_zone = pick(BP_TORSO,BP_TORSO,BP_TORSO,BP_L_LEG,BP_R_LEG,BP_L_ARM,BP_R_ARM,BP_HEAD)
```

Comment:
If I'm reading this right, the spider can bite you in on the hand and inject your face with eggs?

Reply:
What the code here does is it first rolls if it can inject you for that attack.  If it succeeds then it picks a random bodypart and checks if it can inject that targeted bodypart with can_inject(), which checks for robotic/thick material/missing limb/etc.  If it can, it uses get_organ() to get the actual bodypart and then injects it with spider eggs.

Reply:
*shrug, works for me.